modules/event-loop/eventloop/eventloop.go

```diff
@@ -0,0 +1,112 @@
+package eventloop
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+type (
+	MonitorEvent struct {
+		ID          int32
+		Sleep       time.Duration
+		SimulateErr bool
+	}
+
+	AttastatorEvent struct {
+		ID          int32
+		Sleep       time.Duration // Simulate long running work
+		SimulateErr bool
+	}
+
+	Notification struct {
+		Typ string
+		ID  int32
+		Msg string
+		Err error
+	}
+)
+
+func newErrNotification(typ, err string, id int32) Notification {
+	return Notification{
+		Typ: typ,
+		ID:  id,
+		Err: errors.New(err),
+	}
+}
+
+func newMsgNotification(typ, msg string, id int32) Notification {
+	return Notification{
+		Typ: typ,
+		ID:  id,
+		Msg: msg,
+	}
+}
+
+func (n Notification) Error() string {
+	return n.Err.Error()
+}
+
+func (n Notification) String() string {
+	msg := n.Msg
+	if n.Err != nil {
+		msg = n.Error()
+	}
+	return fmt.Sprintf("[%-10s]: [%-3d] %-30s", n.Typ, n.ID, msg)
+}
+
+func Start(ctx context.Context, mCh <-chan MonitorEvent, aCh <-chan AttastatorEvent) <-chan Notification {
```

Comment:
Since we're gonna add another example in the rust code for scheduled queries (instead of incoming events), it would be good to add such an example here too. Ref: https://github.com/cosmos/solidity-ibc-eureka/pull/557/files#r2144953633

Doing this in Go should be the simplest thing in the world with just a couple of tickers instead of events.